routing_model/src/state.rs

```diff
@@ -18,6 +18,7 @@ use crate::utilities::*;
 pub struct CheckAndProcessElderChangeState {
     pub wait_votes: Vec<ParsecVote>,
     pub change_elder: Option<ChangeElder>,
+    pub neighbour_merge_info: Option<NeighbourMergeInfo>,
```

Comment:
As mentioned, should not be there.
You see wait_votes as a couter part WAIT_VOTES.
and change_elder is helpful to implement the update to elder, but is not something that appear in the diagram at all.